scripts/process_diagnostic_messages.rs

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env rust-script
+//!
+//! ```cargo
+//! [dependencies]
+//! serde = { version = "1.0", features = ["derive"] }
+//! serde_json = "1.0"
+//! pathdiff = "0.2.1"
+//! regex = "1.5.4"
+//! ```
+
+use regex::Regex;
+use serde::Deserialize;
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+enum DiagnosticCategory {
+    Error,
+    Message,
+    Suggestion,
+}
+
+#[derive(Clone, Copy, Debug, Deserialize)]
+struct DiagnosticMessageSpec {
+    category: DiagnosticCategory,
+    code: u32,
```

Comment:
It looks like there are some other optional fields on this but I haven't needed them yet so I can presumably just come back to this script and add them as `Option<...>` fields and print them as additional arguments to `diag()` when I need them